vcd/writer.py

```diff
@@ -79,6 +79,7 @@ def __init__(self, file, timescale='1 us', date=None, comment='',
         self._scope_var_names = {}
         self._scope_types = {}
         self._ident_values = OrderedDict()
+        self._init_timestamp = init_timestamp
         self._prev_timestamp = 0
```

Comment:
You are correct. The control flow in `change()` needs to be modified to accommodate. I propose:

        val_str = var.format_value(value, self._check_values)
        ts_int = int(timestamp)

        if ts_int > self._prev_timestamp:
            if self._registering:
                self._finalize_registration()
            if self._dumping:
                print('#', ts_int, sep='', file=self._ofile)
            self._prev_timestamp = ts_int

        if self._dumping and not self._registering:
            print(val_str, file=self._ofile)
        else:
            self._ident_values[var.ident] = val_str

The above passes unit tests for me.

Also, I think we should rename `_prev_timestamp` to just `_timestamp`. This better reflects the semantic of this attribute: it is the `VCDWriter`'s view of the [current] timestamp.